src/mpi/init/netloc_util.c

```diff
@@ -402,6 +402,81 @@ int MPIR_Netloc_get_network_end_point(MPIR_Netloc_network_attributes network_att
     goto fn_exit;
 }
 
+#undef FUNCNAME
+#define FUNCNAME MPIR_Netloc_get_hostnode_index_in_tree
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+int MPIR_Netloc_get_hostnode_index_in_tree(MPIR_Netloc_network_attributes attributes,
+                                           netloc_topology_t topology,
+                                           netloc_node_t * network_endpoint,
+                                           int *index, int *num_leaf_nodes)
+{
+    int mpi_errno = MPI_SUCCESS;
+    int max_level = 0;
+    netloc_node_t **traversal_order;
+    int start_index, end_index, i, j, node_index, num_nodes;
+
+    traversal_order =
+        (netloc_node_t **) MPL_malloc(sizeof(netloc_node_t *) * topology->num_nodes, MPL_MEM_OTHER);
+
+    MPIR_Assert(traversal_order != NULL);
```

Comment:
Asserting is kind of heavy-handed. We have some macros like `MPIR_CHKPMEM_MALLOC` that can help with allocations and return useful error codes. MPICH will still crash in the default case, of course, but we should still attempt to return errors when possible.

Reply:
Used MPIR_CHKPMEM_MALLOC in place of MPL_malloc and assert